### PR TITLE
Create filter sets functionality

### PR DESCRIPTION
**Pull request by jeffmorrisslalom**
This PR enables a user to bring in a new component for the Filter, that has a dropdown with preset filter values. Selecting an option from this dropdown will initialize all of the filters within the Filter Bar with certain default values that the user specifies.
image

### Lakefront PR Checklist
- [ ]  Created new components following the [contributing guide](#https://github.com/ToyotaResearchInstitute/lakefront/blob/main/CONTRIBUTING.md#adding-new-components).
- [ ]  Exported any new components in the `src/index.ts`.
- [ ]  Updated the main [README table](https://github.com/ToyotaResearchInstitute/lakefront#how-to-add-components-to-this-table).
- [x]  Ensure version numbers were bumped properly.
- [ ]  Imported SVGs with relative path, if applicable.
- [ ]  Removed any irrelevant commit messages from merge commit.
- [ ]  Deleted branch after merge.

This closes Issue #276

### CODE REVIEW

File: src/components/Filter/util/filterHooks.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const useFilter = <T extends FilterPostBody>(
     userFilters: FilterSet,
     supportJSON = false,
     location: Location,
-    updateHistory: UpdateHistory
+    updateHistory: UpdateHistory,
```
Comment (alex-maul) [inline src/components/Filter/util/filterHooks.tsx:25]: nit: unneeded ","

File: src/components/Filter/util/filterHooks.tsx [hunk 2]
```diff
@@ -122,6 +127,7 @@ export const useFilter = <T extends FilterPostBody>(
         updateFilter,
         clearFilter,
         clearAllFilters,
-        applyApiPostBody
+        applyApiPostBody,
+        initializePresetValues,
```
Comment (alex-maul) [inline src/components/Filter/util/filterHooks.tsx:131]: nit: unneeded ","

File: src/lib/__tests__/format.test.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        });
+
+        it('will add the word miles to distance other than 1', () => {
+            const dist = 12;
+            const output = addDistanceLabel(dist, units);
+            expect(output).toBe('12 mi');
+        });
+    });
+
+    test('digitGroupSeparator', () => {
```
Comment (alex-maul) [inline src/lib/__tests__/format.test.js:35]: question: was there a reason to use "test" instead of "it" tests?

**Review by alex-maul — APPROVED** [review]

File: src/lib/__tests__/format.test.js (hunk 3 above)
Comment (jeffmorrisslalom) [reply src/lib/__tests__/format.test.js:35]: i updated them to use it now, thanks for catching.

File: src/lib/__tests__/format.test.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
+import {
+    formatNumber,
+    addDistanceLabel,
+    digitGroupSeparator,
+    formatBytes,
+    humanize,
+    pascalToCamelCase,
+    formatLatLong,
+    formatCelsiusTemperature,
+    humanizeCamelCase,
```
[the remark this answers is not in the document]
Comment (jeffmorrisslalom) [reply src/lib/__tests__/format.test.js:10]: done

File: src/components/Filter/util/filterHooks.tsx (hunk 2 above)
Comment (jeffmorrisslalom) [reply src/components/Filter/util/filterHooks.tsx:131]: done

File: src/components/Filter/util/filterHooks.tsx (hunk 1 above)
Comment (jeffmorrisslalom) [reply src/components/Filter/util/filterHooks.tsx:25]: done

File: src/components/Filter/Filter.tsx [hunk 5]
```diff
@@ -108,6 +117,20 @@ export const Filter: FC<FilterComponentProps> = ({
         setActiveSection(newSection);
     };
 
+    const updateFiltersWithPresets = (value: string) => {
+        if (filterMapping) {
+            const filtersToPreset = filterMapping[value];
+            Object.entries(filtersToPreset).forEach(([key, value]) => {
```
Comment (jason-ransom-slalom) [inline src/components/Filter/Filter.tsx:123]: nit: Even though scoping applies/works here, might read a little easier/avoid confusion if you rename one of the `value` argumements (either on line 120 or 123).

File: src/components/Filter/types/index.ts [hunk 6]
```diff
@@ -152,6 +152,11 @@ export interface FilterHooks<T = FilterPostBody> {
      * The function to determine how filters update the post body.
      */
     applyApiPostBody(apiPostBody: T): void;
+
+    /**
+     * The function sets initial filter preset values
```
Comment (jason-ransom-slalom) [inline src/components/Filter/types/index.ts:157]: nit: missing ending period.

File: src/components/Filter/util/__tests__/filterUtil.test.js [hunk 7]
```diff
@@ -3,7 +3,7 @@ import {
     getDefaultJsonViewValue,
     getFilterAppliedCount,
     getUrlFromList,
-    areSetsEqual
+    areSetsEqual, convertToFilterDropdownOptions
```
Comment (jason-ransom-slalom) [inline src/components/Filter/util/__tests__/filterUtil.test.js:6]: nit: add on new line for easier readability.

**Review by jason-ransom-slalom — APPROVED** [review]
LGTM with nit comments.

File: src/components/Filter/Filter.tsx (hunk 5 above)
Comment (jeffmorrisslalom) [reply src/components/Filter/Filter.tsx:123]: oh shoot that's a good call, didn't realize i was using the same variable name.

File: src/components/Filter/types/index.ts (hunk 6 above)
Comment (jeffmorrisslalom) [reply src/components/Filter/types/index.ts:157]: done

File: src/components/Filter/util/__tests__/filterUtil.test.js (hunk 7 above)
Comment (jeffmorrisslalom) [reply src/components/Filter/util/__tests__/filterUtil.test.js:6]: done

**Review by amirkibbar — APPROVED** [review]

**Pull request merged**